src/Npgsql/Replication/ReplicationConnection.cs

```diff
@@ -556,6 +558,9 @@ internal async Task<ReplicationSlotOptions> CreateReplicationSlot(string command
         }
         finally
         {
+            if (columnStream != null && !bypassingStream && !_replicationCancellationTokenSource.Token.IsCancellationRequested)
+                await columnStream.DisposeAsync();
+
```

Comment:
BTW just below in the #if we've got some badly-indented code (that always happens with #if...)

Reply:
Never mind, fixing this in #4646